src/graphics/canvas-layer.js

```diff
@@ -193,7 +222,7 @@ Crafty.canvasLayerObject = {
             for (j = 0, len = q.length; j < len; ++j) {
                 obj = q[j];
 
-                if (dupes[obj[0]] || !obj._visible || !obj.__c.Canvas)
+                if (dupes[obj[0]] || !obj._visible || (obj._drawLayer !== this) )
```

Comment:
The same reason why we previously needed the check that the object has the canvas component -- when we search the map, every entity is returned, not just Canvas entities, and we do the filtering here.

At some point I could see adding configurable hashmaps as a layer option, which would prevent collision checking between layers which opt-in, but would probably allow for several performance improvements.